src/oscar/apps/catalogue/abstract_models.py

```diff
@@ -495,6 +495,16 @@ def options(self):
         pclass_options = self.get_product_class().options.all()
         return set(pclass_options) or set(self.product_options.all())
 
+    @property
```

Comment:
Suggest using `cached_property` here are there are queries inside that we don't want to repeat.